modules/tinymce/src/themes/silver/main/ts/modes/ScrollingContext.ts

```diff
@@ -27,9 +27,13 @@ export const isScroller = (elem: SugarElement<Node> | any): boolean => {
 
 // NOTE: Calculating the list of scrolling ancestors each time this function is called might
 // be unnecessary. It will depend on its usage.
-export const detect = (poupSinkElem: SugarElement<HTMLElement>): Optional<ScrollingContext> => {
-  // We don't want to include popupSinkElem in the list of scrollers, so we just use "ancestors"
-  const scrollers: SugarElement<HTMLElement>[] = PredicateFilter.ancestors(poupSinkElem, isScroller) as SugarElement<HTMLElement>[];
+export const detect = (popupSinkElem: SugarElement<HTMLElement>): Optional<ScrollingContext> => {
+  const ancestorsScrollers = PredicateFilter.ancestors(popupSinkElem, isScroller) as SugarElement<HTMLElement>[];
+
+  const shadowRoot = SugarShadowDom.getShadowRoot(popupSinkElem);
+  const scrollers = ancestorsScrollers.length === 0 && shadowRoot.isSome()
+    ? shadowRoot.map(SugarShadowDom.getShadowHost).map((x) => PredicateFilter.ancestors(x, isScroller)).getOr([]) as SugarElement<HTMLElement>[]
+    : ancestorsScrollers;
```

Comment:
```suggestion
export const detect = (popupSinkElem: SugarElement<HTMLElement>): Optional<ScrollingContext> => {
  const host = SugarShadowDom.getShadowRoot(popupSinkElem).map(SugarShadowDom.getShadowHost).getOr(popupSinkElem);
  const scrollers = PredicateFilter.ancestors(host, isScroller) as SugarElement<HTMLElement>[];
```
If I understand correctly, you are getting the ancestors of either the popupsinkelm or the shadowroot/shadowhost, or the array will be empty. Hope this works in that case.

Reply:
I think Mitch's suggestion should work well from what I can see👍

Reply:
Yeah, I had the same thought initially, but if the scrollable container is within the ShadowRoot, we want to return the element that's within the ShadowRoot, but not beyond that.

Reply:
Would that be the `.getShadowHost` element? In this the host would only either be that element if `getShadowRoot` is Some, or `popupSinkElem` if none. Maybe `PredicateFilter.ancestors` should have a `.getOr([])`?

Reply:
It could be the `.getShadowHost` element or an element outside of the `ShadowHost` and `PredicateFilter.ancestors` cannot traverse beyond the `getShadowRoot`, so I'm trying to retrieve the ShadowHost, then do `PredicateFilter.ancestors` of that ShadowHost element to get the ancestors. `PredicateFilter.ancestors` would return an empty array if the predicate doesn't match.